src/lsrs/helper/tlp-transport.ts

```diff
@@ -0,0 +1,109 @@
+import axios from "axios";
+import {ILogObject, TTransportLogger} from "tslog/src/interfaces";
+
+/**
+ * Domain class representing an error object in the log.
+ */
+class ErrorLog {
+
+    private readonly className: string;
+    private readonly message: string;
+    private readonly stackTrace?: string;
+
+    constructor(error: Error) {
+        this.className = error.constructor.name;
+        this.message = error.message;
+        this.stackTrace = typeof error.stack == "string"
+            ? error.stack
+            : JSON.stringify(error.stack);
+    }
+}
+
+/**
+ * Domain class representing a TLP API compatible log message.
+ */
+class TLPLogMessage {
+
+    private static readonly _SOURCE_APPLICATION = "lsrs";
+
+    private readonly source: string;
+    private readonly threadName: string;
+    private readonly timeStamp: number;
+    private readonly loggerName: string;
+    private readonly level: {levelStr: string};
+    private readonly content: string;
+    private readonly exception?: ErrorLog;
+    
+    /**
+     * This constructor extract and converts log entry data from a TSLog log entry object.
+     *
+     * @param entry TSLog log entry object of type ILogObject
+     */
+    constructor(entry: ILogObject) {
+
+        this.source = TLPLogMessage._SOURCE_APPLICATION;
+        this.threadName = entry.requestId || "main";
+        this.timeStamp = entry.date.getTime();
+        this.loggerName = entry.loggerName ?? "default";
+        this.level = {levelStr: entry.logLevel.toUpperCase()};
+        this.content = entry.argumentsArray[0] as string;
+        this.exception = this.extractException(entry);
+    }
+
+    private extractException(entry: ILogObject): ErrorLog | undefined {
+
+        return entry.argumentsArray.length > 1
```

Comment:
Looks like it's inconvenient to formalize an array of whatevers that we expect to be specific things later down the line. Couldn't check the ILogObject interface, does it make sense to reconsider design at this point or is it fixed?

Reply:
`argumentsArray` contains the log parameters, including the error if passed. And that's it, unfortunately, there's no way in the API to determine specific parts of the log -- I mean other than what I did here. Good news is, I know my own style and the first parameter is always a message, while the second will be an error if appicable. :) Error-prone? Yes. Okay for now? Also yes.